test/assert.js

```diff
@@ -1454,6 +1455,18 @@ describe('assert', function () {
     err(function () {
       assert.property(undefined, 'a', 'blah');
     }, "blah: Target cannot be null or undefined.");
+
+    err(function () {
+      assert.propertyVal(dummyObj, 'a', '2', 'blah');
+    }, "blah: expected { a: '1' } to have property 'a' of '2', but got '1'");
+
+    assert.doesNotThrow(function () {
+      assert.nestedProperty({a:1}, '{a:1}');
+    }, Error, 'the argument to property must be a string when using nested syntax');
+
+    assert.throws(function () {
+      assert.nestedProperty({a:1}, {'a':'1'});
+    }, Error, 'the argument to property must be a string when using nested syntax');
```

Comment:
@solodynamo I'm sorry for making a mess, but I realized that what @meeber meant to suggest is for you to use the `err` function just like on the tests above.

Notice that when you do that it will start to throw `implementation frames not properly filtered from stack trace:`.
Thats because you forgot to pass the `ssfi` argument to the constructor of the `AssertionError`
You are doing this:
```js
throw new AssertionError(flagMsg + 'the argument to property must be a string when using nested syntax');
```
But you need to do this:
```js
throw new AssertionError(
  flagMsg + 'the argument to property must be a string when using nested syntax',
  undefined,
  ssfi
);
```

I'm sorry again for the confusion I made.

Reply:
Also, please pass a third argument to test if the custom message works.
Your test should look like this:
```js
err(function () {
  assert.nestedProperty({a:1}, {'a':'1'}, 'blah');
}, 'blah: the argument to property must be a string when using nested syntax');
```